src/model/tree/DataSourceTreeNode.ts

```diff
@@ -62,31 +61,18 @@ export class DataSourceTreeNode extends DVTreeItem {
 	}
 
 	isValueType(): boolean {
-		return !this.isEmpty() &&
-			!utils.isConfigMapRef(this.dataSourceConfig.properties[0].valueFrom) &&
-			!utils.isSecretRef(this.dataSourceConfig.properties[0].valueFrom);
+		return true;
 	}
 
 	isSecretType(): boolean {
-		return !this.isEmpty() &&
-			utils.isSecretRef(this.dataSourceConfig.properties[0].valueFrom);
+		return false;
 	}
 
 	isConfigMapType(): boolean {
-		return !this.isEmpty() &&
-			utils.isConfigMapRef(this.dataSourceConfig.properties[0].valueFrom);
+		return false;
 	}
 
 	getReferenceName(): string {
```

Comment:
shouldn't we completely remove the getReferenceName from this class now as there is a specific one for handling reference?

Reply:
Removing it will complicate other code because I then need to check what type the TreeNode actually is (a reference or a normal one). This way I don't need to take care because both node classes have that function.